packages/ListBox/src/components/Filter/Filter.js

```diff
@@ -67,30 +61,38 @@ const Filter = React.forwardRef((props, ref) => {
     applyFilter(dispatch, applyFilterType)(filteredOptions, noResultsFound);
   };
 
-  const handleBlur = () => {
-    window.requestAnimationFrame(() => {
-      if (hasFooter) {
-        // if has footer the responsible of closing is the Footer no the onblur event
-        return;
-      }
-
-      if (document.activeElement !== refListBoxContainer.current) {
-        // this will reset the activeOption and close the Popover
+  React.useEffect(() => {
+    let id = null;
+    if (!isOpen) {
+      id = window.requestAnimationFrame(() => {
```

Comment:
Don't see a reason why `id` is defined outside of the `if` statement: 

```js
if (!isOpen) {
  const id = indow.requestAnimationFrame(() => {
```

Reply:
or, maybe the cleanup function need to be outside of the `if ` statement instead 🤷

Reply:
do you mean because will hoist and then the clear function will have access to it?

I thought this way is clear that the value will be used as well on the clear function. or do you mean something else?